app/models/educator.rb

```diff
@@ -34,85 +34,26 @@ def as_json(options = {})
-
-  def admin_gets_access_to_all_students
-    errors.add(:admin, "needs access to all students") if admin? && !has_access_to_all_students?
-  end
-
-  def grade_level_access_is_array_of_strings
-    unless grade_level_access.all? { |grade| grade.instance_of? String }
-      errors.add(:grade_level_access, "should be an array of strings")
-    end
-  end
-
-  def grade_level_strings_are_valid
-    if grade_level_access.any? { |grade| !(VALID_GRADES.include?(grade)) }
-      errors.add(:grade_level_access, "invalid grade")
-    end
-  end
-
-  def grade_level_strings_are_uniq
-    unless grade_level_access.uniq.size == grade_level_access.size
-      errors.add(:grade_level_access, "duplicate values")
-    end
-  end
-
-  def grade_level_access_is_not_nil
-    errors.add(:grade_level_access, "cannot be nil") if grade_level_access.nil?
-  end
-
-  # This method is the source of truth for whether an educator is authorized to view information about a particular
-  # student.
   def is_authorized_for_student(student)
-    return true if self.districtwide_access?
-
-    return false if self.restricted_to_sped_students && !(student.program_assigned.in? ['Sp Ed', 'SEIP'])
-    return false if self.restricted_to_english_language_learners && student.limited_english_proficiency == 'Fluent'
-    return false if self.school_id.present? && self.school_id != student.school_id
-
-    return true if self.schoolwide_access? || self.admin? # Schoolwide admin
-    return true if self.has_access_to_grade_levels? && student.grade.in?(self.grade_level_access) # Grade level access
-    return true if student.in?(self.students) # Homeroom level access
-    return true if student.in?(self.section_students) # Section level access
-    false
+    Authorizer.new(self).is_authorized_for_student?(student)
   end
 
-  def is_authorized_for_school(currentSchool)
-    self.districtwide_access? ||
-    (self.schoolwide_access? && self.school == currentSchool) ||
-    (self.has_access_to_grade_levels? && self.school == currentSchool)
+  def is_authorized_for_school(current_school)
+    Authorizer.new(self).is_authorized_for_school?(current_school)
   end
 
   def is_authorized_for_section(section)
-    return true if self.districtwide_access?
-
-    return false if self.school.present? && self.school != section.course.school
-
-    return true if self.schoolwide_access? || self.admin?
-    return true if section.in?(self.sections)
-    false
+    Authorizer.new(self).is_authorized_for_section?(section)
   end
 
   def students_for_school_overview(*additional_includes)
-    return [] unless school.present?
-
-    if schoolwide_access?
-      school.students
-            .active
-            .includes(additional_includes || [])
-    elsif has_access_to_grade_levels?
-      school.students
-            .active
-            .where(grade: self.grade_level_access)
-            .includes(additional_includes || [])
-    else
+    students = Authorizer.new(self).students_for_school_overview
```

Comment:
This is the only substantive change - it removes the if branch here and lets `Authorizer` handle that.